src/internal/connector/onedrive/collections.go

```diff
@@ -14,94 +14,89 @@ import (
 	"github.com/alcionai/corso/src/internal/observe"
 	"github.com/alcionai/corso/src/pkg/logger"
 	"github.com/alcionai/corso/src/pkg/path"
-	"github.com/alcionai/corso/src/pkg/selectors"
 )
 
-// Collections is used to retrieve OneDrive data for a
-// specified user
+type driveSource int
```

Comment:
I'm curious abouth the choice of using a const to switch execution flow instead of defining an interface and passing a concrete instance of a struct that implements the interface. It looks like right now the main things this const is used for is getting the path of the item and getting the drives. The sharepoint and onedrive versions of each of these functions take the same arguments as well so it seems like an interface would work no? If using the interface, the calling pattern would be slightly reversed I guess where the caller would make the instance of the implementation and pass it in?

Reply:
Great question.  What I like about enumeration is that the code (which are ultimately minor pivots in behavior) remained a bit more consolidated.  Given that there's little actual difference between the processes, it felt like a lighter touch.  But no strong feelings here.  I can swap it out for interfaces if that's preferred.

Reply:
It seems like we have a combination of both approaches here. We're passing in an enum to switch execution flow for path computation (`GetCanonicalPath`) but an interface to pass in the matcher.

Reply:
I suppose I don't see that combination as an anti-pattern.  But it sounds like he preference is for an interface, so I'll swap out for that.

Reply:
I don't think it's an anti pattern - so go with what seems right and easy to read. We're in nit territory here 😃

Reply:
Going to hold off on the interface for now.  Since onedrive is generating NewCollections from connector, not inside onedrive, the supply of interface-compliant structs is getting a bit messy.  I can revisit this later, as it might be easier with some of the other proposed refactors.